framework/include/base/EigenProblem.h

```diff
@@ -41,15 +41,30 @@ class EigenProblem : public FEProblemBase
   virtual bool converged() override;
 
   virtual unsigned int getNEigenPairsRequired() { return _n_eigen_pairs_required; }
+  virtual void setNEigenPairsRequired(unsigned int n_eigen_pairs)
+  {
+    _n_eigen_pairs_required = n_eigen_pairs;
+  }
   virtual bool isGeneralizedEigenvalueProblem() { return _generalized_eigenvalue_problem; }
   virtual bool isNonlinearEigenvalueSolver();
 
+  bool residualInitialed() { return _is_residual_initialed; }
+  void setResidualInitialed(bool is_residual_initialed)
```

Comment:
Can you document this method? I'm not sure what it's for, it seems odd to me that you can initialize what I would assume is internal state of the object from a method call. Why is this needed?